pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -586,12 +601,45 @@ def getData(self):
                     y1[:,0] = y2.max(axis=1)
                     y1[:,1] = y2.min(axis=1)
                     y = y1.reshape(n*2)
-                
-                    
+
+            if self.opts['dynamicRangeLimit'] is not None:
+                view_range = self.viewRect()
+                if view_range is not None:
+                    data_range = self.dataRect()
+                    if data_range is not None:
+                        view_height = view_range.height()
+                        lim = self.opts['dynamicRangeLimit']
+                        if data_range.height() > lim * view_height:
+                            min_val = view_range.top()    - lim * view_height
+                            max_val = view_range.bottom() + lim * view_height
+                            y = np.clip(y, a_min=min_val, a_max=max_val)
+
             self.xDisp = x
             self.yDisp = y
         return self.xDisp, self.yDisp
 
+    def dataRect(self):
+        """
+        Returns a bounding rectangle (as QRectF) for the full set of data.
+        Will return None if there is no data or if all values (x or y) are NaN.
+        """
+        if self._dataRect is not None:
+            return self._dataRect
+        if self.xData is None or self.yData is None:
+            return None
+        ymin = np.nanmin(self.yData)
```

Comment:
we should catch the runtime warning being emitted here as we account for the `nan` case just below.

This can be done with the following code

```python
import warnings

...

with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    np.nanmin(self.yData)
```